azkaban-common/src/main/java/azkaban/executor/ExecutorManager.java

```diff
@@ -58,38 +60,53 @@
  *
  */
 public class ExecutorManager extends EventHandler implements
-    ExecutorManagerAdapter {
+  ExecutorManagerAdapter {
+  static final String AZKABAN_USE_MULTIPLE_EXECUTORS =
+    "azkaban.use.multiple.executors";
   private static Logger logger = Logger.getLogger(ExecutorManager.class);
-  private ExecutorLoader executorLoader;
-  private String executorHost;
-  private int executorPort;
+  final private ExecutorLoader executorLoader;
 
-  private CleanerThread cleanerThread;
+  final private CleanerThread cleanerThread;
 
-  private ConcurrentHashMap<Integer, Pair<ExecutionReference, ExecutableFlow>> runningFlows =
-      new ConcurrentHashMap<Integer, Pair<ExecutionReference, ExecutableFlow>>();
-  private ConcurrentHashMap<Integer, ExecutableFlow> recentlyFinished =
-      new ConcurrentHashMap<Integer, ExecutableFlow>();
+  final private ConcurrentHashMap<Integer, Pair<ExecutionReference, ExecutableFlow>> runningFlows =
+    new ConcurrentHashMap<Integer, Pair<ExecutionReference, ExecutableFlow>>();
+  final private ConcurrentHashMap<Integer, ExecutableFlow> recentlyFinished =
+    new ConcurrentHashMap<Integer, ExecutableFlow>();
 
-  private ExecutingManagerUpdaterThread executingManager;
+  /* map to easily access queued flows */
+  final private ConcurrentHashMap<Integer, Pair<ExecutionReference, ExecutableFlow>> queuedFlowMap =
+    new ConcurrentHashMap<Integer, Pair<ExecutionReference, ExecutableFlow>>();
+  /* web server side queue */
+  final private ConcurrentLinkedQueue<Pair<ExecutionReference, ExecutableFlow>> queuedFlows =
```

Comment:
maybe just need better naming to distinguish this member variable and the one above it.